bundles/org.openhab.binding.dominoswiss/src/main/java/org/openhab/binding/dominoswiss/internal/EGateHandler.java

```diff
@@ -0,0 +1,332 @@
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EgateHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Frieso Aeschbacher - Initial contribution
+ */
+@NonNullByDefault
+public class EGateHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EGateHandler.class);
+    private @Nullable Socket egateSocket;
+
+    private int port;
+    private @Nullable String host;
+    private static final int SOCKET_TIMEOUT_SEC = 250;
+    private final Object lock = new Object();
+    private @Nullable BufferedWriter writer;
+    private @Nullable BufferedReader reader;
+    private @Nullable Future<?> refreshJob;
+    private @Nullable Map<String, ThingUID> registeredBlinds;
+    private final ScheduledExecutorService scheduler;
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    public EGateHandler(Bridge thing) {
+        super(thing);
+        registeredBlinds = new HashMap<String, ThingUID>();
+        scheduler = (ScheduledExecutorService) Executors
+                .newSingleThreadExecutor(new NamedThreadFactory(thing.getUID().getAsString(), true));
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(GETCONFIG)) {
+            sendCommand("EthernetGet;\r");
+        }
+    }
+
+    @Override
+    public void initialize() {
+        Configuration config;
+        config = this.getConfig();
+        host = (String) config.get("ipAddress");
+        port = 1318; // default Port
+        if (!config.get("port").toString().isEmpty()) {
+            port = Integer.parseInt(config.get("port").toString()); // take configured Port
+        }
+
+        if (host != null && port > 0) {
+            // Create a socket to eGate
+            pollingJob = scheduler.scheduleWithFixedDelay(pollingConfig, 0, 30, TimeUnit.SECONDS);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Cannot connect to dominoswiss eGate gateway. host IP address or port are not set.");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        try {
+            egateSocket.close();
+            refreshJob.cancel(true);
+            reader.close();
+            if (pollingJob != null) {
+                pollingJob.cancel(true);
+            }
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED, "EGate Disposed");
+            logger.debug("EGate Handler connection closed, disposing");
+        } catch (IOException e) {
+            logger.error("EGate Handler Error on dispose: {} ", e.toString());
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. This could be debug. See this link for a description of the log levels: https://www.openhab.org/docs/developer/guidelines.html#f-logging

Please check all.